### PR TITLE
s/dsl/rewriter/

### PR DESCRIPTION
**Pull request by ptarjan**
I was stuck on some other work so I thought I'd see if this was easy. Turns out it was.

### Motivation
This pass barely does DSLs nowadays. It is just a bunch of syntax transforms. I'm totally open to other names if you want to suggest any. The renames are all done programmatically so aren't hard to change the names.

We deiced in our standup to name it Rewriter 🎉 

Some decisions I made:
* Instead of having `patchDSL` and `replaceDSL` methods, they are now just `run` methods like the other passes
* It is named in the singular like `name-tree` as `rewrite-tree`
* I left the arg named `dsl-plugins` since I think I want to encourage those to be limited (like we originally hoped for our own dsl pass)
* I didn't work too hard to keep the `|`s aligned in the docs since alignment is the enemy of renames

I'd suggest just reviewing the commit names. Anything named `replace` is done with https://github.com/ptarjan/dotfiles/blob/master/bin/replace, and the others are just unix commands

### Test plan

See included automated tests.

### CODE REVIEW

File: docs/internals.md [hunk 1]
```diff
@@ -627,7 +627,7 @@ See [core/Symbols.h] and [core/SymbolRef.h] for more information.
 <!-- Phase descriptions -->
 [parser]: #parser
 [desugar]: #desugar
-[dsl]: #dsl
+[dsl]: #syntax_transform
```
Comment (aisamanra) [inline docs/internals.md]: Tiny nit: this `dsl` should probably change, too.

Comment (ptarjan) [reply docs/internals.md]: great catch!

File: main/options/options.h [hunk 2]
```diff
@@ -93,7 +93,7 @@ enum Phase {
     INIT,
     PARSER,
     DESUGARER,
-    DSL,
+    SyntaxTransform,
```
Comment (aisamanra) [inline main/options/options.h]: This also should probably be `SYNTAX_TRANSFORM` to match style with the rest of this enum.

File: syntax_transform/ClassNew.cc [hunk 3]
```diff
@@ -1,16 +1,16 @@
...
-#include "dsl/dsl.h"
+#include "syntax_transform/syntax_transform.h"
 
 using namespace std;
 
-namespace sorbet::dsl {
+namespace sorbet::syntax_transform {
 
-vector<unique_ptr<ast::Expression>> ClassNew::replaceDSL(core::MutableContext ctx, ast::Assign *asgn) {
+vector<unique_ptr<ast::Expression>> ClassNew::run(core::MutableContext ctx, ast::Assign *asgn) {
```
Comment (jez) [inline syntax_transform/ClassNew.cc]: I liked that we could distinguish between different kinds of DSL passes before. patchDSL / replaceDSL are not both conceptually `run` in my mind. I'm in general much happier writing `replaceDSL` DSL passes than `patchDSL` passes when possible.

**Comment by ptarjan** [comment]
sure, I can keep the distinction. What would you name them in the new world?

File: syntax_transform/ClassNew.cc (hunk 3 above)
Comment (ptarjan) [reply syntax_transform/ClassNew.cc]: sure, I can keep the distinction. What would you name them in the new world?

**Pull request merged**